physics/SFC_Models/Land/Noahmp/lnd_iau_mod.F90

```diff
@@ -0,0 +1,879 @@
+         write(6,*) trim(ioerrmsg)         
+         errmsg = 'lnd_iau_mod_set_control: error reading namelist file '//trim(fn_nml)  &
+                  // 'the error message from file handler:' //trim(ioerrmsg) 
+         errflg = 1
+         return
+       end if
+   endif 
+#endif
+ 
+888 if (ios /= 0) then  ! .and. ios /= iostat_end) then
+         write(iosstr, '(I0)') ios
+         if (me == mpi_root) then
+           write(6,*) 'lnd_iau_mod_set_control: I/O error code '//trim(iosstr)//' at land_iau namelist read' 
+         endif         
+         errmsg = 'lnd_iau_mod_set_control: I/O error code '//trim(iosstr)//' at land_iau namelist read'  
+         errflg = 1
+         return
+    end if 
+       
+999 if (ios /= 0) then   ! ios .eq. iostat_end) then  
+        write(iosstr, '(I0)') ios
+        if (me == mpi_root) then
+          WRITE(6, * ) 'lnd_iau_mod_set_control: Warning! EoF ('//trim(iosstr)//') while reading land_iau namelist,' &
+                  // ' likely because land_iau_nml was not found in input.nml. It will be set to default.' 
+        endif
+   endif
+
+   if (me == mpi_root) then
+      write(6,*) "land_iau_nml"
+      write(6, land_iau_nml)
+   endif
+   
+   Land_IAU_Control%do_land_iau = do_land_iau
+   Land_IAU_Control%iau_delthrs = land_iau_delthrs
+   Land_IAU_Control%iau_inc_files = land_iau_inc_files
+   Land_IAU_Control%iaufhrs = land_iau_fhrs   
+   Land_IAU_Control%iau_filter_increments = land_iau_filter_increments
+   Land_IAU_Control%lsoil_incr = lsoil_incr
+
+   Land_IAU_Control%me = me
+   Land_IAU_Control%mpi_root = mpi_root
+   Land_IAU_Control%isc = isc
+   Land_IAU_Control%jsc = jsc
+   Land_IAU_Control%nx = nx
+   Land_IAU_Control%ny = ny
+   Land_IAU_Control%tile_num = tile_num
+   Land_IAU_Control%nblks = nblks
+   Land_IAU_Control%lsoil = lsoil
+   Land_IAU_Control%lsnow_lsm = lsnow_lsm
+   Land_IAU_Control%dtp = dtp
+   Land_IAU_Control%fhour = fhour
+
+!   Land_IAU_Control%input_nml_file = input_nml_file
+!   Land_IAU_Control%input_nml_file_length = input_nml_file_length
+
+   Land_IAU_Control%upd_stc = land_iau_upd_stc
+   Land_IAU_Control%upd_slc = land_iau_upd_slc
+   Land_IAU_Control%do_stcsmc_adjustment = land_iau_do_stcsmc_adjustment
+   Land_IAU_Control%min_T_increment = land_iau_min_T_increment
+
+   allocate(Land_IAU_Control%blksz(nblks))
+   allocate(Land_IAU_Control%blk_strt_indx(nblks))
+
+   ! Land_IAU_Control%blk_strt_indx: start index of each block, for flattened (ncol=nx*ny) arrays 
+   ! required in noahmpdriv_run to get subsection of the stc array for each
+   ! proces/thread
+   ix = 1
+   do nb=1, nblks
+      Land_IAU_Control%blksz(nb) = blksz(nb)
+      Land_IAU_Control%blk_strt_indx(nb) = ix
+      ix = ix + blksz(nb)
+   enddo
+
+end subroutine land_iau_mod_set_control
+
+subroutine land_iau_mod_init (Land_IAU_Control, Land_IAU_Data, errmsg, errflg)     !nlunit, ncols, IPD_Data,,Init_parm)
+   ! integer,                              intent(in) :: me, mpi_root

```

Comment:
Is there any good reason to keep the commented out variable declarations?

Reply:
> Is there any good reason to keep the commented out variable declarations?

not anymore.